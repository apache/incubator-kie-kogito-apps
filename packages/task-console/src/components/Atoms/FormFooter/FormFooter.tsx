import React from 'react';
import { ActionGroup, Button } from '@patternfly/react-core';
import { FormActionDescription } from '../../../model/FormDescription';

interface IOwnProps {
  actions?: FormActionDescription[];
  onActionClick: (action: FormActionDescription) => void;
}

const FormFooter: React.FC<IOwnProps> = ({ actions, onActionClick }) => {
  return actions && actions.length > 0 ? (
    <ActionGroup>
      {actions.map(action => {
        return (
          <Button
            key={'submit-' + action.name}
            type="submit"
            variant={action.primary ? 'primary' : 'secondary'}
            onClick={() => onActionClick(action)}
          >
            {action.name}
          </Button>
        );
      })}
    </ActionGroup>
  ) : null;
};

export default FormFooter;
