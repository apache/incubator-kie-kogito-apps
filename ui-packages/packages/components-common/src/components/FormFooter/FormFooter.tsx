/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import _ from 'lodash';
import { ActionList, ActionListItem } from '@patternfly/react-core';
import { convertActionsToButton, FormAction } from '../utils';
import { componentOuiaProps, OUIAProps } from '@kogito-apps/ouia-tools';
import '../styles.css';

interface IOwnProps {
  actions?: FormAction[];
  enabled?: boolean;
  onSubmitForm?: () => void;
}

const FormFooter: React.FC<IOwnProps & OUIAProps> = ({
  actions,
  enabled = true,
  onSubmitForm,
  ouiaId,
  ouiaSafe
}) => {
  if (_.isEmpty(actions)) {
    return null;
  }

  const actionItems = convertActionsToButton(
    actions,
    enabled,
    onSubmitForm
  ).map((button, index) => {
    return (
      <ActionListItem key={`form-action-${index}`}>{button}</ActionListItem>
    );
  });

  return (
    <div
      className="kogito-components-common__form-footer-padding-top"
      {...componentOuiaProps(ouiaId, 'form-footer', ouiaSafe)}
    >
      <ActionList>{actionItems}</ActionList>
    </div>
  );
};

export default FormFooter;
