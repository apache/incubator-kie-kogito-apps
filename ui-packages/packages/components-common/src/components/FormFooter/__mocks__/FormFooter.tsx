import React from 'react';

const MockedFormFooter = (): React.ReactElement => {
  return <></>;
};

export default MockedFormFooter;
