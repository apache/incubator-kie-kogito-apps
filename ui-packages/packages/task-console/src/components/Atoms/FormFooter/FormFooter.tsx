import React from 'react';
import { ActionGroup, Button } from '@patternfly/react-core';

export interface IFormAction {
  name: string;
  primary?: boolean;
  onActionClick: () => void;
}

interface IOwnProps {
  actions?: IFormAction[];
}

const FormFooter: React.FC<IOwnProps> = ({ actions }) => {
  return actions && actions.length > 0 ? (
    <ActionGroup>
      {actions.map(action => {
        return (
          <Button
            key={'submit-' + action.name}
            type="submit"
            variant={action.primary ? 'primary' : 'secondary'}
            onClick={action.onActionClick}
          >
            {action.name}
          </Button>
        );
      })}
    </ActionGroup>
  ) : null;
};

export default FormFooter;
