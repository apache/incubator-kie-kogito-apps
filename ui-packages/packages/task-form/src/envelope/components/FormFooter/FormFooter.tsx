/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import _ from 'lodash';
import { ActionList, ActionListItem } from '@patternfly/react-core';
import { convertActionsToButton, FormAction } from '../utils';
import { componentOuiaProps, OUIAProps } from '@kogito-apps/components-common';

interface IOwnProps {
  actions?: FormAction[];
  enabled?: boolean;
}

const FormFooter: React.FC<IOwnProps & OUIAProps> = ({
  actions,
  enabled = true,
  ouiaId,
  ouiaSafe
}) => {
  if (_.isEmpty(actions)) {
    return null;
  }

  const actionItems = convertActionsToButton(actions, enabled).map(
    (button, index) => {
      return (
        <ActionListItem key={`form-action-${index}`}>{button}</ActionListItem>
      );
    }
  );

  return (
    <ActionList {...componentOuiaProps(ouiaId, 'form-footer', ouiaSafe)}>
      {actionItems}
    </ActionList>
  );
};

export default FormFooter;
