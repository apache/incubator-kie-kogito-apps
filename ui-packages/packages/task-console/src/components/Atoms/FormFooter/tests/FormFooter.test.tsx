import React from 'react';
import { mount } from 'enzyme';
import FormFooter, { IFormAction } from '../FormFooter';

describe('Form Footer test', () => {
  it('testing showing actions', () => {
    const actions: IFormAction[] = [
      {
        name: 'action1',
        primary: true,
        onActionClick: jest.fn()
      },
      {
        name: 'action2',
        primary: false,
        onActionClick: jest.fn()
      }
    ];

    const wrapper = mount(<FormFooter actions={actions} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('testing showing empty actions', () => {
    const props = {
      actions: []
    };

    const wrapper = mount(<FormFooter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('testing showing no actions', () => {
    const wrapper = mount(<FormFooter />);
    expect(wrapper).toMatchSnapshot();
  });

  it('testing action click', () => {
    const action1 = {
      name: 'action1',
      primary: true,
      onActionClick: jest.fn()
    };

    const action2 = {
      name: 'action2',
      primary: false,
      onActionClick: jest.fn()
    };

    const props = {
      actions: [action1, action2]
    };

    const wrapper = mount(<FormFooter {...props} />);
    expect(wrapper).toMatchSnapshot();

    const button1 = wrapper.findWhere(node => node.key() === 'submit-action1');
    button1.simulate('click');

    expect(action1.onActionClick).toBeCalledTimes(1);

    const button2 = wrapper.findWhere(node => node.key() === 'submit-action2');
    button2.simulate('click');

    expect(action2.onActionClick).toBeCalledTimes(1);
  });
});
