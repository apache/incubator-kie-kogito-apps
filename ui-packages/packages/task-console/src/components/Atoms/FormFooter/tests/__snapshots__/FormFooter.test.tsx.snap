// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Footer test testing action click 1`] = `
<FormFooter
  actions={
    Array [
      Object {
        "name": "action1",
        "onActionClick": [MockFunction],
        "primary": true,
      },
      Object {
        "name": "action2",
        "onActionClick": [MockFunction],
        "primary": false,
      },
    ]
  }
>
  <ActionGroup>
    <div
      className="pf-c-form__group pf-m-action"
    >
      <div
        className="pf-c-form__actions"
      >
        <Component
          key="submit-action1"
          onClick={[MockFunction]}
          type="submit"
          variant="primary"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "action1",
                "onClick": [MockFunction],
                "type": "submit",
                "variant": "primary",
              }
            }
            consumerContext={null}
          >
            <Button
              onClick={[MockFunction]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              type="submit"
              variant="primary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                disabled={false}
                onClick={[MockFunction]}
                tabIndex={null}
                type="submit"
              >
                action1
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
        <Component
          key="submit-action2"
          onClick={[MockFunction]}
          type="submit"
          variant="secondary"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "action2",
                "onClick": [MockFunction],
                "type": "submit",
                "variant": "secondary",
              }
            }
            consumerContext={null}
          >
            <Button
              onClick={[MockFunction]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              type="submit"
              variant="secondary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-secondary"
                disabled={false}
                onClick={[MockFunction]}
                tabIndex={null}
                type="submit"
              >
                action2
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
      </div>
    </div>
  </ActionGroup>
</FormFooter>
`;

exports[`Form Footer test testing showing actions 1`] = `
<FormFooter
  actions={
    Array [
      Object {
        "name": "action1",
        "onActionClick": [MockFunction],
        "primary": true,
      },
      Object {
        "name": "action2",
        "onActionClick": [MockFunction],
        "primary": false,
      },
    ]
  }
>
  <ActionGroup>
    <div
      className="pf-c-form__group pf-m-action"
    >
      <div
        className="pf-c-form__actions"
      >
        <Component
          key="submit-action1"
          onClick={[MockFunction]}
          type="submit"
          variant="primary"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "action1",
                "onClick": [MockFunction],
                "type": "submit",
                "variant": "primary",
              }
            }
            consumerContext={null}
          >
            <Button
              onClick={[MockFunction]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              type="submit"
              variant="primary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-primary"
                disabled={false}
                onClick={[MockFunction]}
                tabIndex={null}
                type="submit"
              >
                action1
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
        <Component
          key="submit-action2"
          onClick={[MockFunction]}
          type="submit"
          variant="secondary"
        >
          <ComponentWithOuia
            component={[Function]}
            componentProps={
              Object {
                "children": "action2",
                "onClick": [MockFunction],
                "type": "submit",
                "variant": "secondary",
              }
            }
            consumerContext={null}
          >
            <Button
              onClick={[MockFunction]}
              ouiaContext={
                Object {
                  "isOuia": false,
                  "ouiaId": null,
                }
              }
              type="submit"
              variant="secondary"
            >
              <button
                aria-disabled={null}
                aria-label={null}
                className="pf-c-button pf-m-secondary"
                disabled={false}
                onClick={[MockFunction]}
                tabIndex={null}
                type="submit"
              >
                action2
              </button>
            </Button>
          </ComponentWithOuia>
        </Component>
      </div>
    </div>
  </ActionGroup>
</FormFooter>
`;

exports[`Form Footer test testing showing empty actions 1`] = `
<FormFooter
  actions={Array []}
/>
`;

exports[`Form Footer test testing showing no actions 1`] = `<FormFooter />`;
